syncer/syncer.go

```diff
@@ -151,36 +157,37 @@ func (s *Syncer) IsSynced(ctx context.Context) bool {
 
 // Start starts the main sync loop that tries to sync data for every SyncInterval.
 func (s *Syncer) Start(ctx context.Context) {
-	s.syncOnce.Do(func() {
-		if s.ticker.GetCurrentLayer().Uint32() <= 1 {
-			s.setSyncState(ctx, synced)
-		}
-		for {
-			select {
-			case <-s.shutdownCtx.Done():
-				s.logger.WithContext(ctx).Info("stopping sync to shutdown")
-				return
-			case <-s.syncTimer.C:
-				s.synchronize(ctx)
-			}
+	s.syncOnce.Do(
+		func() {
+			s.tomb.Go(s.loop)
+		})
+}
+
+func (s *Syncer) loop() error {
+	if s.ticker.GetCurrentLayer().Uint32() <= 1 {
+		s.setSyncState(s.tomb.Context(nil), synced)
+	}
+	ctx := s.tomb.Context(nil)
+	for {
+		select {
+		case <-s.tomb.Dying():
+			s.logger.WithContext(ctx).Info("stopping sync to shutdown")
+			return nil
+		case <-s.syncTimer.C:
+			s.synchronize(ctx)
 		}
-	})
+	}
 }
 
 // ForceSync manually starts a sync process outside the main sync loop. If the node is already running a sync process,
 // ForceSync will be ignored.
 func (s *Syncer) ForceSync(ctx context.Context) {
 	s.logger.WithContext(ctx).Debug("executing ForceSync")
-	go s.synchronize(ctx)
-}
-
-func (s *Syncer) isClosed() bool {
-	select {
-	case <-s.shutdownCtx.Done():
-		return true
-	default:
-		return false
-	}
+	// synchronize only allows one process to proceed so no need to worry about spawning too many goroutines
+	s.tomb.Go(func() error {
```

Comment:
this is probably not in the scope of this PR, but maybe it should pass a notification that will be handled by the loop?

Reply:
i think the original idea was that if this RPC call comes in during the a synchronize run, it will be ignored instead of being processed after the current run.
i don't see a huge advantage one way over the other.